### PR TITLE
Change editor viewport mode tracker id from viewport to entity context.

### PR DESCRIPTION
**Pull request by jonawals**
**Overview**

This PR changes the id associated with each `ViewportEditorModeTracker` from `ViewportId` to `EntityContxtId`. The implication here is that trackers are now organized by entity context and not viewport.

**Testing**

```
[==========] Running 31 tests from 5 test cases.
[----------] Global test environment set-up.
[----------] 4 tests from ViewportEditorModesTestsFixture
[ RUN      ] ViewportEditorModesTestsFixture.NumberOfEditorModesIsEqualTo4
[       OK ] ViewportEditorModesTestsFixture.NumberOfEditorModesIsEqualTo4 (0 ms)
[ RUN      ] ViewportEditorModesTestsFixture.InitialEditorModeStateHasAllInactiveModes
[       OK ] ViewportEditorModesTestsFixture.InitialEditorModeStateHasAllInactiveModes (1 ms)
[ RUN      ] ViewportEditorModesTestsFixture.SettingOutOfBoundsModeActiveReturnsError
[       OK ] ViewportEditorModesTestsFixture.SettingOutOfBoundsModeActiveReturnsError (0 ms)
[ RUN      ] ViewportEditorModesTestsFixture.SettingOutOfBoundsModeInactiveReturnsError
[       OK ] ViewportEditorModesTestsFixture.SettingOutOfBoundsModeInactiveReturnsError (0 ms)
[----------] 4 tests from ViewportEditorModesTestsFixture (1 ms total)

[----------] 6 tests from ViewportEditorModeTrackerTestFixture
[ RUN      ] ViewportEditorModeTrackerTestFixture.InitialCentralStateTrackerHasNoViewportEditorModess
[       OK ] ViewportEditorModeTrackerTestFixture.InitialCentralStateTrackerHasNoViewportEditorModess (480 ms)
[ RUN      ] ViewportEditorModeTrackerTestFixture.ActivatingViewportEditorModeForNonExistentIdCreatesViewportEditorModesForThatId
[       OK ] ViewportEditorModeTrackerTestFixture.ActivatingViewportEditorModeForNonExistentIdCreatesViewportEditorModesForThatId (292 ms)
[ RUN      ] ViewportEditorModeTrackerTestFixture.DeactivatingViewportEditorModeForNonExistentIdCreatesViewportEditorModesForThatIdButReturnsError
[       OK ] ViewportEditorModeTrackerTestFixture.DeactivatingViewportEditorModeForNonExistentIdCreatesViewportEditorModesForThatIdButReturnsError (298 ms)
[ RUN      ] ViewportEditorModeTrackerTestFixture.GettingNonExistentViewportEditorModesForIdReturnsNull
[       OK ] ViewportEditorModeTrackerTestFixture.GettingNonExistentViewportEditorModesForIdReturnsNull (265 ms)
[ RUN      ] ViewportEditorModeTrackerTestFixture.ActivatingViewportEditorModesForExistingIdInThatStateReturnsError
[       OK ] ViewportEditorModeTrackerTestFixture.ActivatingViewportEditorModesForExistingIdInThatStateReturnsError (251 ms)
[ RUN      ] ViewportEditorModeTrackerTestFixture.DeactivatingViewportEditorModesForExistingIdNotInThatStateReturnssError
[       OK ] ViewportEditorModeTrackerTestFixture.DeactivatingViewportEditorModesForExistingIdNotInThatStateReturnssError (296 ms)
[----------] 6 tests from ViewportEditorModeTrackerTestFixture (1892 ms total)

[----------] 2 tests from ViewportEditorModePublisherTestFixture
[ RUN      ] ViewportEditorModePublisherTestFixture.ActivatingViewportEditorModesForExistingIdPublishesOnViewportEditorModeActivateEventForAllSubscribers
[       OK ] ViewportEditorModePublisherTestFixture.ActivatingViewportEditorModesForExistingIdPublishesOnViewportEditorModeActivateEventForAllSubscribers (248 ms)
[ RUN      ] ViewportEditorModePublisherTestFixture.DeactivatingViewportEditorModesForExistingIdPublishesOnViewportEditorModeDeactivatingEventForAllSubscribers
[       OK ] ViewportEditorModePublisherTestFixture.DeactivatingViewportEditorModesForExistingIdPublishesOnViewportEditorModeDeactivatingEventForAllSubscribers (250 ms)
[----------] 2 tests from ViewportEditorModePublisherTestFixture (503 ms total)

[----------] 3 tests from ViewportEditorModeTrackerIntegrationTestFixture
[ RUN      ] ViewportEditorModeTrackerIntegrationTestFixture.InitialViewportEditorModeIsDefault
[       OK ] ViewportEditorModeTrackerIntegrationTestFixture.InitialViewportEditorModeIsDefault (250 ms)
[ RUN      ] ViewportEditorModeTrackerIntegrationTestFixture.EnteringComponentModeAfterInitialStateHasViewportEditorModesDefaultAndComponentModeActive
[       OK ] ViewportEditorModeTrackerIntegrationTestFixture.EnteringComponentModeAfterInitialStateHasViewportEditorModesDefaultAndComponentModeActive (243 ms)
[ RUN      ] ViewportEditorModeTrackerIntegrationTestFixture.EnteringEditorPickEntitySelectionAfterInitialStateHasOnlyViewportEditorModePickModeActive
[       OK ] ViewportEditorModeTrackerIntegrationTestFixture.EnteringEditorPickEntitySelectionAfterInitialStateHasOnlyViewportEditorModePickModeActive (251 ms)
[----------] 3 tests from ViewportEditorModeTrackerIntegrationTestFixture (750 ms total)

[----------] 16 tests from AllEditorModes/ViewportEditorModesTestsFixtureWithParams
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/0
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/0 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/1
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/1 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/2
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/2 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/3
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeActiveActivatesOnlyThatMode/3 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/0
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/0 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/1
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/1 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/2
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/2 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/3
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingModeInactiveInactivatesOnlyThatMode/3 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/0
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/0 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/1
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/1 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/2
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/2 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/3
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesActiveActivatesAllThoseModesNonMutuallyExclusively/3 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/0
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/0 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/1
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/1 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/2
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/2 (0 ms)
[ RUN      ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/3
[       OK ] AllEditorModes/ViewportEditorModesTestsFixtureWithParams.SettingMultipleModesInactiveInactivatesAllThoseModesNonMutuallyExclusively/3 (0 ms)
[----------] 16 tests from AllEditorModes/ViewportEditorModesTestsFixtureWithParams (29 ms total)

[----------] Global test environment tear-down
[==========] 31 tests from 5 test cases ran. (3186 ms total)
[  PASSED  ] 31 tests.
```

Signed-off-by: John <jonawals@amazon.com>

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h [hunk 1]
```diff
@@ -23,11 +23,11 @@ namespace AzToolsFramework
         Pick
     };
 
-    //! Viewport identifier and other relevant viewport data.
+    //! Viewport editor mode tracker identifier and other relevant data.
     struct ViewportEditorModeInfo
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h:27]: question: Should this be renamed to `ViewportEditorModeTracker` (or `ViewportEditorModeTrackerInfo`) just going from the comment update?

File: Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.cpp [hunk 2]
```diff
@@ -218,7 +218,7 @@ namespace AzToolsFramework
             // this call to activate the component mode editor state should eventually replace the bus call in
             // ComponentModeCollection::BeginComponentMode() to EditorComponentModeNotifications::EnteredComponentMode
             // such that all of the notifications for activating/deactivating the different editor modes are in a central location
-            m_viewportEditorModeTracker->ActivateMode({ /* DefaultViewportId */ }, ViewportEditorMode::Component);
+            m_viewportEditorModeTracker->ActivateMode({ GetEntityContextId() }, ViewportEditorMode::Component);
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.cpp:221]: comment: In future we'll ideally want to pass this through (ideally when `ComponentModeCollection` gets created we pass in the `EntityContextId` from the context that created it) but this is a bigger change and makes sense to happen as a separate PR, just wanted to mention it for future work 🙂

File: Code/Framework/AzToolsFramework/Tests/Viewport/ViewportEditorModeTests.cpp [hunk 3]
```diff
@@ -158,10 +154,13 @@ namespace UnitTest
 
         void SetUpEditorFixtureImpl() override
         {
+            m_handlerIds.resize(ViewportEditorModes::NumEditorModes);
             for (auto mode = 0; mode < ViewportEditorModes::NumEditorModes; mode++)
             {
-                m_editorModeHandlers[mode] = AZStd::make_unique<ViewportEditorModeNotificationsBusHandler>(mode);
+                m_handlerIds[mode] = TrackerId::CreateRandom();
+                m_editorModeHandlers[mode] = AZStd::make_unique<ViewportEditorModeNotificationsBusHandler>(m_handlerIds[mode]);
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/Tests/Viewport/ViewportEditorModeTests.cpp:161]: comment: A guide comment might be useful here just for some context for the reader

**Review by hultonha — APPROVED** [review]
LGTM! Thanks for updating this 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h [hunk 4]
```diff
@@ -23,11 +23,11 @@ namespace AzToolsFramework
...
     {
-        using IdType = AzFramework::ViewportId;
-        IdType m_id = ViewportUi::DefaultViewportId; //!< The unique identifier for a given viewport.
+        using IdType = AzFramework::EntityContextId;
+        IdType m_id = AzFramework::EntityContextId::CreateNull(); //!< The unique identifier for a given viewport editor mode tracker.
```
Comment (AMZN-daimini) [inline Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h:30]: nit: comment still says viewport

**Review by AMZN-daimini — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h (hunk 1 above)
Comment (jonawals) [reply Code/Framework/AzToolsFramework/AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h:27]: fixed 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.cpp (hunk 2 above)
Comment (jonawals) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.cpp:221]: :+1

File: Code/Framework/AzToolsFramework/Tests/Viewport/ViewportEditorModeTests.cpp (hunk 3 above)
Comment (jonawals) [reply Code/Framework/AzToolsFramework/Tests/Viewport/ViewportEditorModeTests.cpp:162]: fixed 👍

**Pull request merged**